### ISSUE TITLE
Configuration system

### ISSUE THREAD

**Issue by doublevcodes** [doublevcodes/sirius#4]
The user needs to be able to set options within a `sirius.config.toml` file. This will be implemented using `tomli`

**Comment by Shivansh-007** [comment]
I would like to work on this, it would be based on the work done by arl on discord-modmail/modmail#75, where one could write configuration classes using marshmallow and attrs. This would support auto-generation of docs **in** the config file.

Example:
```py
@attr.s(auto_attribs=True, slots=True)
class TwitterAuthCfg:
    """Twitter account authentication configuration."""

    api_key: str = attr.ib(
        default="",
        on_setattr=attr.setters.frozen,
        repr=False,
        metadata={
            "required": True,
            "allow_none": False,
            METADATA_TABLE: ConfigMetadata(
                description="Think of this as the user name that represents your account when using the app.",
            ),
        },
    )
    api_secret: str = attr.ib(
        default="",
        on_setattr=attr.setters.frozen,
        repr=False,
        metadata={
            "required": True,
            "allow_none": False,
            METADATA_TABLE: ConfigMetadata(
                description="Think of this as the password that represents your account when using the app.",
            ),
        },
    )

```

Would auto-generate:
```toml
[auth]
# Think of this as the user name that represents your account when using the app.
api_key = ""
# Think of this as the password that represents your account when using the app.
api_secret = ""
```

**Comment by doublevcodes** [comment]
So how would this work, a default configuration is generated with lots of empty keys? I'd prefer that the configuration file only contain keys that need to be used.

**Comment by Shivansh-007** [comment]
> So how would this work, a default configuration is generated with lots of empty keys? I'd prefer that the configuration file only contain keys that need to be used.

Yes, it would only include the config keys which are required, and possibly I could add a separate flag if the user wants to generate with "all" of the keys.

**Comment by doublevcodes** [comment]
> > So how would this work, a default configuration is generated with lots of empty keys? I'd prefer that the configuration file only contain keys that need to be used.
> 
> 
> 
> Yes, it would only include the config keys which are required, and possibly I could add a separate flag if the user wants to generate with "all" of the keys.

That sounds good.

So:

- Configuration powered by marshmallow and attrs
- A new CLI command to bootstrap a project with an initial configuration

**Comment by Shivansh-007** [comment]
Should be closed by #6
